Entity/BaseMenuItem.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Grossum\MenuBundle\Entity;
+
+/**
+ * BaseMenuItem
+ */
+abstract class BaseMenuItem
+{
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var \DateTime
+     */
+    protected $createdAt;
+
+    /**
+     * @var \DateTime
+     */
+    protected $updatedAt;
+
+    /**
+     * @var BaseMenu
+     */
+    protected $menu;
+
+    /**
+     * Set name
+     *
+     * @param string $name
+     *
+     * @return BaseMenuItem
```

Comment:
Please replace all annotations `@return BaseMenuItem` to `@return $this`. This allow to use autocomplete in class-children